python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/experimental/test_resources.py

```diff
@@ -144,3 +158,226 @@ def test_basic_resource_request(
     assert_rest_api_call(call=api_calls[3], endpoint="jobs", object_id=TEST_CONNECTION_ID)
     assert_rest_api_call(call=api_calls[4], endpoint=f"jobs/{TEST_JOB_ID}")
     assert_rest_api_call(call=api_calls[5], endpoint=f"jobs/{TEST_JOB_ID}")
+
+
+@pytest.mark.parametrize(
+    "status",
+    [
+        AirbyteJobStatusType.SUCCEEDED,
+        AirbyteJobStatusType.CANCELLED,
+        AirbyteJobStatusType.ERROR,
+        AirbyteJobStatusType.FAILED,
+        TEST_UNRECOGNIZED_AIRBYTE_JOB_STATUS_TYPE,
+    ],
+    ids=[
+        "job_status_succeeded",
+        "job_status_cancelled",
+        "job_status_error",
+        "job_status_failed",
+        "job_status_unrecognized",
+    ],
+)
+def test_airbyte_sync_and_poll_client_job_status(
+    status: str, base_api_mocks: responses.RequestsMock
+) -> None:
+    resource = AirbyteCloudWorkspace(
+        workspace_id=TEST_WORKSPACE_ID,
+        client_id=TEST_CLIENT_ID,
+        client_secret=TEST_CLIENT_SECRET,
+    )
+    client = resource.get_client()
+
+    test_job_endpoint = f"jobs/{TEST_JOB_ID}"
+    test_job_api_url = f"{AIRBYTE_REST_API_BASE}/{AIRBYTE_REST_API_VERSION}/{test_job_endpoint}"
+
+    # Create mock responses to mock full sync and poll behavior to test statuses, used only in this test
+    base_api_mocks.add(
+        method=responses.POST,
+        url=f"{AIRBYTE_REST_API_BASE}/{AIRBYTE_REST_API_VERSION}/jobs",
+        json=get_job_details_sample(status=AirbyteJobStatusType.PENDING),
+        status=200,
+    )
+    base_api_mocks.add(
+        method=responses.GET,
+        url=test_job_api_url,
+        json=get_job_details_sample(status=status),
+        status=200,
+    )
+    base_api_mocks.add(
+        method=responses.POST,
+        url=f"{AIRBYTE_CONFIGURATION_API_BASE}/{AIRBYTE_CONFIGURATION_API_VERSION}/connections/get",
+        json=SAMPLE_CONNECTION_DETAILS,
+        status=200,
+    )
+
+    if status == TEST_UNRECOGNIZED_AIRBYTE_JOB_STATUS_TYPE:
+        base_api_mocks.add(
+            method=responses.DELETE,
+            url=test_job_api_url,
+            status=200,
+            json=get_job_details_sample(status=AirbyteJobStatusType.CANCELLED),
+        )
```

Comment:
confused by this

Reply:
We have another test dedicated to `cancel_on_termination`, so let's remove this part of the test when testing statuses - responses.RequestsMock yells if a mocked request is not called in a test, so it must be added only when the job is cancelled.